src/runtime/audience-action-flow.js

```diff
@@ -266,12 +266,38 @@ export class AudienceActionFlow {
   // eslint-disable-next-line no-unused-vars
   handleSurveyDataTransferRequest_(request) {
     // @TODO(justinchou): execute callback with setOnInterventionComplete
-    // and Google Analytics, then check for success
+    // then check for success
+    const gaLoggingSuccess = this.logSurveyDataToGoogleAnalytics(request);
     const surveyDataTransferResponse = new SurveyDataTransferResponse();
-    surveyDataTransferResponse.setSuccess(true);
+    surveyDataTransferResponse.setSuccess(gaLoggingSuccess);
     this.activityIframeView_.execute(surveyDataTransferResponse);
   }
 
+  /**
+   * Logs SurveyDataTransferRequest to Google Analytics. Returns boolean
+   * for whether or not logging was successful.
+   * @param {SurveyDataTransferRequest} request
+   * @return {boolean}
+   * @private
+   */
+  logSurveyDataToGoogleAnalytics(request) {
```

Comment:
@oyj9109 do we not want to use the event messaging system to send the data instead of sending to GA directly from here? Trying to think about future use-cases where they use their own endpoint instead of GA to capture survey information

Reply:
Having a direct access to the GA function here is preferred as it's guaranteed the function is available and actually invoked. If we delegate it to event manager, it's hard to handle it via async flow.
Even though some publishers implement their own endpoint and provides callbacks in the future, I think we'll still support GA-based data collection so this will be still beneficial.

Reply:
Yeah, we have an internal bug around being able to ensure something captured the event from the event manager, but I bring it up because the current logic **requires** GA to be set up. We should make sure that:
1) Publishers know they need to have it
2) We may not even want to show the survey on the page if they dont. I'd rather push a warning to the console indicating it is required vs failing this late in the flow

Reply:
Definitely! Great call outs.
1. In experiment, we'll clearly communicate this through. They should know it because it's the only way they can collect data. In the future, we'll need to make it clear in the publisher configuration flow. (either callback or GA is required)
2. Seems like Justin already started working on it. See https://github.com/subscriptions-project/swg-js/pull/2421

Reply:
+1 on refactor

Reply:
Please see new changes!